sat/cli/bootprep/input/configuration.py

```diff
@@ -365,3 +374,62 @@ def get_cfs_api_data(self):
                                            f'of configuration {self.name}')
 
         return cfs_api_data
+
+    def create(self, payload):
+        """Create the CFS configuration with a request to the CFS API.
+
+        Args:
+            payload (dict): the payload to pass to the CFS API to create the
+                CFS configuration
+        """
+        try:
+            # request_body guaranteed to have 'name' key, so no need to catch ValueError
+            self.cfs_client.put_configuration(self.name, payload)
+        except APIError as err:
+            raise ConfigurationCreateError(f'Failed to create configuration: {err}')
+
+
+class InputConfigurationCollection(BaseInputItemCollection):
+    """The collection of CFS configurations defined in the input file."""
+
+    item_class = InputConfiguration
+
+    def __init__(self, items_data, instance, jinja_env, request_dumper, cfs_client, **kwargs):
+        """Create a new InputConfigurationCollection.
+
+        Args:
+            items_data (list of dict): CFS configuration data from input file,
+                already validated by schema
+            instance (sat.bootprep.input.InputInstance): a reference to the
+                full instance loaded from the config file
+            jinja_env (jinja2.Environment): the Jinja2 environment in which
+                fields supporting Jinja2 templating should be rendered.
+            request_dumper (sat.cli.bootprep.output.RequestDumper): the dumper
+                for dumping request data to files.
+            cfs_client (csm_api_client.service.cfs.CFSClient): the CFS API client
+            **kwargs: additional keyword arguments
+        """
+        super().__init__(items_data, instance, jinja_env, request_dumper,
+                         cfs_client=cfs_client, **kwargs)
+        self.cfs_client = cfs_client
+
+    def get_existing_items_by_name(self):
+        """Get existing CFS configurations by name.
+
+        See parent class for full docstring.
+        """
+        try:
+            # TODO: Add a get_configurations method to cfs_client?
+            configurations = self.cfs_client.get('configurations').json()
+        except APIError as err:
+            # TODO: Consider whether we need subclasses of InputItemCreateError
```

Comment:
My impression so far is that we don't but especially not for configurations with this refactor.

It's mildly confusing that there are subclasses for `ConfigurationCreateError` and `ImageCreateError` but not one for session templates, even though right now `InputConfiguration` and `InputSessionTemplate` are the two that are subclasses of `BaseInputItem`.

Reply:
Yeah, I agree it is confusing. I will go ahead an replace all usages of `ConfigurationCreateError` with just `InputItemCreateError`.  We can probably do the same for `ImageCreateError` to simplify if/when that gets refactored too.